Labyrinth/src/LabyrinthOriginal/MainPanel.java

```diff
@@ -0,0 +1,27 @@
+package LabyrinthOriginal;
+
+import java.awt.BorderLayout;
+
+import javax.swing.JPanel;
+
+import Physics.BoardDisplayer;
+
+public class MainPanel extends JPanel {
+
+	/**
+	 * I just used the default here... I don't really know much about serialization yet.
+	 */
+	private static final long serialVersionUID = 1L;
+	
+	private BoardDisplayer boardDisplayer;
+
+	public MainPanel(BoardDisplayer bd) {
+		this.boardDisplayer = bd;
+		this.setLayout(new BorderLayout());
+		this.add(this.boardDisplayer, BorderLayout.CENTER);
+	}
+	
+	public BoardDisplayer getBoardDisplayer() {
```

Comment:
I would like to remove this method, because I think it is something that people should try to avoid using. If anyone needs to mess with the main panel's board displayer, it should either 1. have kept a reference to the bd when it was passed in, or 2. use a wrapper method written in this class, like `MainPanel.doSomethingWithYourTiles`

Reply:
Ok! Sounds good to me.